engine/src/test/scala/cromwell/webservice/PartialWorkflowSourcesSpec.scala

```diff
@@ -1,17 +1,49 @@
 package cromwell.webservice
 
+import cats.data.Validated.{Invalid, Valid}
 import org.scalatest.{FlatSpec, Matchers}
-import spray.json._
 import spray.json.DefaultJsonProtocol._
+import spray.json._
 
 class PartialWorkflowSourcesSpec extends FlatSpec with Matchers {
-  it should "succesfully merge and override multiple input files" in {
+
+  behavior of "mergeMaps method"
+  it should "successfully merge and override multiple input files" in {
     val input1 = Map("wf.a1" -> "hello", "wf.a2" -> "world").toJson.toString
     val input2 = Map.empty[String, String].toJson.toString
     val overrideInput1 = Map("wf.a2" -> "universe").toJson.toString
-    val allInputs = PartialWorkflowSources.mergeMaps(Seq(Option(input1), Option(input2), Option(overrideInput1)))
+    val mergedMapsErrorOr = PartialWorkflowSources.mergeMaps(Seq(Option(input1), Option(input2), Option(overrideInput1)))
+
+    mergedMapsErrorOr.isValid shouldBe true
+
+    mergedMapsErrorOr.map(inputs => {
+      inputs.fields.keys should contain allOf("wf.a1", "wf.a2")
+      inputs.fields("wf.a2") should be(JsString("universe"))
+    })
+  }
+
+  it should "return error when workflow input is not a valid json object" in {
+    val invalidJsonInput = "\"invalidInput\""
+    val mergedMapsErrorOr = PartialWorkflowSources.mergeMaps(Seq(Option(invalidJsonInput)))
+
+    mergedMapsErrorOr.isValid shouldBe false
+
+    mergedMapsErrorOr match {
+      case Valid(_) => "This is unexpected! This test is designed to fail!"
+      case Invalid(error) => error.head shouldBe "Submitted input '\"invalidInput\"' belongs to class spray.json.JsString, which is not a valid json object."
+    }
+  }
+
+  it should "return error when auxiliary workflow input is not a valid json" in {
+    val validInput = "{\"key\":\"value\"}"
+    val invalidAuxJsonInput = "invalidInput"
+    val mergedMapsErrorOr = PartialWorkflowSources.mergeMaps(Seq(Option(validInput), Option(invalidAuxJsonInput)))
+
+    mergedMapsErrorOr.isValid shouldBe false
 
-    allInputs.fields.keys should contain allOf("wf.a1", "wf.a2")
-    allInputs.fields("wf.a2") should be(JsString("universe"))
+    mergedMapsErrorOr match {
+      case Valid(_) => "This is unexpected! This test is designed to fail!"
```

Comment:
wrap this around a `fail`: `fail("This is unexpected! This test is designed to fail!")` to make the test actually fail. As it is now the test would still pass if it got here

Reply:
TIL. Thank you!